vaadin-upload-flow-parent/vaadin-upload-flow/src/main/java/com/vaadin/flow/component/upload/Upload.java

```diff
@@ -493,6 +505,517 @@ protected void fireUpdateProgress(long totalBytes, long contentLength) {
+            ComponentEventListener<UploadRequestEvent> listener) {
+        return addListener(UploadRequestEvent.class,
+                (ComponentEventListener) listener);
+    }
+
+    @DomEvent("upload-response")
+    public static class UploadResponseEvent extends ComponentEvent<Upload> {
+        private final JsonObject detail;
+        private final JsonObject detailXhr;
+        private final JsonObject detailFile;
+
+        public UploadResponseEvent(Upload source, boolean fromClient,
+                @EventData("event.detail") JsonObject detail,
+                @EventData("event.detail.xhr") JsonObject detailXhr,
+                @EventData("event.detail.file") JsonObject detailFile) {
+            super(source, fromClient);
+            this.detail = detail;
+            this.detailXhr = detailXhr;
+            this.detailFile = detailFile;
+        }
+
+        public JsonObject getDetail() {
+            return detail;
+        }
+
+        public JsonObject getDetailXhr() {
+            return detailXhr;
+        }
+
+        public JsonObject getDetailFile() {
+            return detailFile;
+        }
+    }
+
+    /**
+     * Adds a listener for {@code upload-response} events fired by the
+     * webcomponent.
+     *
+     * @param listener
+     *            the listener
+     * @return a {@link Registration} for removing the event listener
+     */
+    @SuppressWarnings({ "rawtypes", "unchecked" })
+    protected Registration addUploadResponseListener(
+            ComponentEventListener<UploadResponseEvent> listener) {
+        return addListener(UploadResponseEvent.class,
+                (ComponentEventListener) listener);
+    }
+
+    @DomEvent("upload-retry")
+    public static class UploadRetryEvent extends ComponentEvent<Upload> {
+        private final JsonObject detail;
+        private final JsonObject detailXhr;
+        private final JsonObject detailFile;
+
+        public UploadRetryEvent(Upload source, boolean fromClient,
+                @EventData("event.detail") JsonObject detail,
+                @EventData("event.detail.xhr") JsonObject detailXhr,
+                @EventData("event.detail.file") JsonObject detailFile) {
+            super(source, fromClient);
+            this.detail = detail;
+            this.detailXhr = detailXhr;
+            this.detailFile = detailFile;
+        }
+
+        public JsonObject getDetail() {
+            return detail;
+        }
+
+        public JsonObject getDetailXhr() {
+            return detailXhr;
+        }
+
+        public JsonObject getDetailFile() {
+            return detailFile;
+        }
+    }
+
+    /**
+     * Adds a listener for {@code upload-retry} events fired by the
+     * webcomponent.
+     *
+     * @param listener
+     *            the listener
+     * @return a {@link Registration} for removing the event listener
+     */
+    @SuppressWarnings({ "rawtypes", "unchecked" })
+    protected Registration addUploadRetryListener(
```

Comment:
Didn't find any use for this method, could be removed

Reply:
Removed.